### PR TITLE
Surge with gRPC / Side Cars POC

### PR DESCRIPTION
**Pull request by sebastianharko**

### CODE REVIEW

**Review by garyKeorkunian — COMMENTED** [review]
Looks good so far.  Some comments below.

File: modules/sidecar-poc/src/main/protobuf/messages.proto [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+message HandleEventRequest {
+  State state = 1;
+  Event event = 2;
+}
+
+message HandleEventResponse {
+  State state = 1;
+}
+
+service Service {
```
[the remark this answers is not in the document]
Comment (sebastianharko) [reply modules/sidecar-poc/src/main/protobuf/messages.proto:38]: @garyKeorkunian Yes, you are right, and this initial POC probably will not have perfect naming etc. trying to validate everything from end to end and once everything is validated we can iterate on things like finding the best naming for the Service.

File: modules/sidecar-poc/src/main/scala/com/example/Main.scala [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import surge.core.command.AggregateCommandModelCoreTrait
+import surge.kafka.KafkaTopic
+import surge.scaladsl.command.{AggregateCommandModel, SurgeCommand, SurgeCommandBusinessLogic}
+
+import java.util.UUID
+import scala.concurrent.Await
+import scala.util.{Failure, Success, Try}
+import scala.concurrent.duration._
+
+object GenericCommandModel extends AggregateCommandModel[State, Command, Event] {
```
[the remark this answers is not in the document]
Comment (sebastianharko) [reply modules/sidecar-poc/src/main/scala/com/example/Main.scala]: yes you are correct, it's unnecessary

File: modules/sidecar-poc/src/main/protobuf/messages.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (sebastianharko) [reply modules/sidecar-poc/src/main/protobuf/messages.proto]: @garyKeorkunian Fixed the name btw ;-)

**Comment by jeffboutotte** [comment]
@sebastianharko is this PR needed anymore? Looks like a lot of the functionality was merged in with #80 and #97

**Pull request closed without merging**